File: src/api/RiskAnalytics.Api.Repository/Models/MonthlyUtilization.cs
namespace RiskAnalytics.Api.Repository.Models;

public class MonthlyUtilization
{
    public string? Group { get; set; } = null!;
    public string? AircraftType { get; set; }
    public int? GroupId { get; set; }
    public int NumberOfAircraftInGroup { get; set; }

    public int Year { get; set; }

    public int Month { get; set; }

    public double TotalHours { get; set; }
    public double AverageHours { get; set; }
    public int NumberOfAircraftWithHours { get; set; }

    public double TotalCycles { get; set; }
    public double AverageCycles { get; set; }
    public int NumberOfAircraftWithCycles { get; set; }

    public double AverageHoursPerCycle { get; set; }
    public int NumberOfAircraftWithHoursPerCycle { get; set; }

    public double TotalCo2KgPerSeat { get; set; }
    public double AverageCo2KgPerSeat { get; set; }
    public int NumberOfAircraftWithCo2KgPerSeat { get; set; }

    public double TotalCo2GPerAsk { get; set; }
    public double AverageCo2GPerAsk { get; set; }
    public int NumberOfAircraftWithCo2GPerAsk { get; set; }

    public double TotalCo2GPerAsm { get; set; }
    public double AverageCo2GPerAsm { get; set; }
    public int NumberOfAircraftWithCo2GPerAsm { get; set; }
}

File: src/frontend/src/app/modules/compare-portfolio/models/monthly-utilization.ts
export interface MonthlyUtilization {
  group: string;
  groupId: number;
  aircraftType: string;
  numberOfAircraftInGroup: number;
  year: number;
  month: number;
  averageHours: number;
  totalHours: number;
  numberOfAircraftWithHours: number;
  averageCycles: number;
  totalCycles: number;
  numberOfAircraftWithCycles: number;
  averageHoursPerCycle: number;
  numberOfAircraftWithHoursPerCycle: number;
  averageCo2KgPerSeat: number;
  numberOfAircraftWithCo2KgPerSeat: number;
  totalCo2KgPerSeat: number;
  averageCo2GPerAsk: number;
  averageCo2GPerAsm: number;
  totalCo2GPerAsk: number;
  totalCo2GPerAsm: number;
  numberOfAircraftWithCo2GPerAsk: number;
  numberOfAircraftWithCo2GPerAsm: number;
}
